apps/wallet-mobile/src/yoroi-wallets/utils/utils.ts

```diff
@@ -133,22 +133,31 @@ export const Quantities = {
 
     return absoluteQuantity.isEqualTo(minimalFractionalPart)
   },
-  parseFromText: (text: string, denomination: number, format: NumberLocale) => {
+  parseFromText: (text: string, denomination: number, format: NumberLocale, precision = denomination) => {
     const {decimalSeparator} = format
     const invalid = new RegExp(`[^0-9${decimalSeparator}]`, 'g')
     const sanitized = text === '' ? '' : text.replaceAll(invalid, '')
     if (sanitized === '') return ['', `0`] as [string, Balance.Quantity]
     if (sanitized.startsWith(decimalSeparator)) return [`0${decimalSeparator}`, `0`] as [string, Balance.Quantity]
     const parts = sanitized.split(decimalSeparator)
+
+    let fullDecValue = sanitized
+    let value = sanitized
+
     const isDec = parts.length >= 2
 
-    const fullDecValue = isDec ? `${parts[0]}${decimalSeparator}${parts[1].slice(0, denomination)}1` : sanitized
+    if (isDec) {
+      const [int, dec] = parts
+      // trailing `1` is to allow user type `1.0` without losing the decimal part
+      fullDecValue = `${int}${decimalSeparator}${dec.slice(0, precision)}1`
+      value = `${int}${decimalSeparator}${dec.slice(0, precision)}`
+    }
+
     const fullDecFormat = new BigNumber(fullDecValue.replace(decimalSeparator, '.')).toFormat()
     const input = isDec ? fullDecFormat.slice(0, -1) : fullDecFormat
```

Comment:
move inside the if, add a comment that the trailing is removed inside the hasDecimal logic

Reply:
You'll have to declare input as `let` before and move both lines inside the if

Which is why I preferred everything as const and the ternaries 😅

Reply:
BigNumber is object so const/let won't make that difference cuz you still can mutate it, but I also prefer and agree on not changing after declaring with initialization. To look even better I'd move everything dec / not decimal and return based on that, we will have some duplication, but readability is more important duplication.

Reply:
Maybe it's because I wrote it, but it looked more clear to me before. Now with the lets, an if dividing everything, the early return and a couple lines duplicated because of it, looks harder to follow to me.